src/colorizer/utils/analytics_utils.ts

```diff
@@ -0,0 +1,47 @@
+import { FeatureType } from "../Dataset";
+
+// Added by Google Tag Manager.
+// Custom events are triggered by pushing to the `event` field.
+// Additional data can also be pushed to include it as a parameter in the event.
+// Note that data will persist in the dataLayer until the page is reloaded;
+// this means that certain fields (like datasetWriterVersion) can be included in all subsequent
+// events without needing to be re-pushed.
+declare global {
+  interface Window {
+    // Optional because it is undefined in testing environments
+    dataLayer?: Record<string, any>[];
+  }
+}
+
+export const enum AnalyticsEvent {
+  COLLECTION_LOAD = "collection_load",
+  DATASET_LOAD = "dataset_load",
+  EXPORT_COMPLETE = "export_complete",
+  FEATURE_SELECTED = "feature_selected",
+  ROUTE_ERROR = "route_error",
+}
```

Comment:
do these have to be defined at the google analytics back end?  Or does it just take whatever we send?

Reply:
Yes! Well. 
From what I understand, GA will still log the events, but they won't trigger the corresponding tag that actually captures and then does interesting stuff with that information (e.g. set it up for reports/exploration). So yes, all of these have counterparts in GA Tag Manager that I've set up.